### ISSUE TITLE
[Bug]: adding a share-link breaks encryption after upgrade to v27

### ISSUE THREAD

**Issue by dsweberr** [nextcloud/server#39429]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

adding a share via link to existing files breaks encryption by deleting all encryption keys

https://github.com/nextcloud/server/blob/v27.0.0/apps/encryption/lib/Crypto/Encryption.php#L444
$this->path is empty and causes exception after deleting all keys in #L441

changing to $path appears to fix the issue

### Steps to reproduce

enable encryption, do multiple major version upgrades, add share via link to file

### Expected behavior

link to file should work and not break existing files

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Logs

```shell
"exception": {
        "Exception": "BadMethodCallException",
        "Message": "path needs to be relative to the system wide data folder and point to a user specific file",
        "Code": 0,
        "Trace": [
            {
                "file": "/var/www/nextcloud/lib/private/Encryption/Keys/Storage.php",
                "line": 366,
                "function": "getUidAndFilename",
                "class": "OC\\Encryption\\Util",
                "type": "->"
            },
            {
                "file": "/var/www/nextcloud/lib/private/Encryption/Keys/Storage.php",
                "line": 138,
                "function": "getFileKeyDir",
                "class": "OC\\Encryption\\Keys\\Storage",
                "type": "->"
            },
            {
                "file": "/var/www/nextcloud/apps/encryption/lib/KeyManager.php",
                "line": 381,
                "function": "setFileKey",
                "class": "OC\\Encryption\\Keys\\Storage",
                "type": "->"
            },
            {
                "file": "/var/www/nextcloud/apps/encryption/lib/Crypto/Encryption.php",
                "line": 444,
                "function": "setShareKey",
                "class": "OCA\\Encryption\\KeyManager",
                "type": "->"
            },
---
call to setShareKey() with empty path
```

**Comment by dsweberr** [comment]
appears to be fixed in
https://github.com/nextcloud/server/commit/d7ef87cc9545c3e30e669200ff98b8c17ff7d388

**Comment by joshtrichards** [comment]
Fixed in #39447
